### DISCUSSION TITLE
Nowy interfejs

### DISCUSSION THREAD

**Discussion by ookris** [hackerspace-silesia/Smogomierz#138] [Ideas]
Trochę mi zeszło ale pierwsza wersja   (a w zasadzie to już kolejna miniwersja) już jest i jakoś nawet działa. Choć czasami nie wszystkie tłumaczenia ładują się prawidłowo. Czasami zostają 2-3 nieprzetłumaczone opcji w {}. Czasami też strona konfiguracyjna nie chce się załadować w ogóle. 

**Zmiany:**
- FW oparte na wersji: **2.6.2** 
- Całość oparta jest o framework [Bootstrap](https://getbootstrap.com) w wersji 4 
- Strona konfiguracyjna ma "stary wygląd" - nowy styl powodował problemy z generowaniem strony
- Strona z pomiarami ma nowy wygląd. Docelowo cały interfejs będzie miał taki styl
- Zwiększono wysokość wykresów przez co są teraz bardziej czytelne

**Do zrobienia:**

- [ ] Przystosowanie strony konfiguracyjnej oraz strony aktualizacji. 
- [ ] Poprawki w tłumaczeniu 

Gotowe firmware:
_Smogomierz_2.6.2.UI_PMS_build_28.03.2019.bin_: https://drive.google.com/open?id=1kArTuYEnVG-kT5URhXMOpUSvrYy1aMd0

**Comment by bfaliszek** [comment]
Zostaje kilka {}, bo kończy się pamięć w ESP. Postaram się w wolnej chwili trochę zoptymalizować jej zużycie, żeby coś zaoszczędzić i nie było problemów z tłumaczeniem 😉

**Comment by hejsiri** [comment]
Pokażesz jak wyglada to nowe GUI?

**Comment by ookris** [comment]
Całość będzie nieco szersza i jaśniejsza. Kilka rzeczy trzeba jeszcze ogarnąć ale ogólny zarys już jest. 
![Zrzut ekranu 2019-03-24 o 17 52 32](https://user-images.githubusercontent.com/26855867/54882758-a34a6e80-4e5d-11e9-8285-4481669c5565.png)
![Zrzut ekranu 2019-03-24 o 17 48 33](https://user-images.githubusercontent.com/26855867/54882759-a34a6e80-4e5d-11e9-9dca-49b7855f9941.png)

**Comment by krupniok200** [comment]
powiem że strona w takim stylu jest dużo czytelniejsza, szczególnie strona z pomiarami, konfiguracja czy update ważne aby działała :)

**Comment by hejsiri** [comment]
Fajnie, że podświetla pomiar kolorem. Konfiguracja nie chce się u mnie załadować, wykresy zrobiłbym wyższe razy 2.

**Comment by ookris** [comment]
@mientki problem z ładowaniem strony konfiguracyjnej wynika, jak twierdzi @bfaliszek, z kończącej się pamięci w ESP. Wykresy można zrobić wyższe 2x ale w przypadku kiedy włączymy wyświetlanie wszystkich strona zrobi się nieco dłuższa i żeby dotrzeć do interesujących wykresów trzeba będzie sporo przewijać. Mógłbym porobić zakładki dla każdego z serwisów generujących wykresy ale nie wiem czy ESP, które jest już nieco przeładowane, to wytrzyma.

**Comment by bfaliszek** [comment]
ESP8266 ma 80KB pamięci dla zmiennych. Normalnie zajęte jest jakieś 63% z tego. Problem robi się, gdy wjeżdża tłumaczenie. Tam jest sporo Stringów, które zajmują pamięć. Gdy się kończy to zostają nieprzetłumaczone {}. Jak będę miał chwilę, to muszę spróbować zmienić te Stringi na char, żeby zaoszczędzić pamięci.

**Comment by hejsiri** [comment]
Optymalizować :) osobiście wróciłem do oryginału bo u mnie ma po prostu działać zawsze.

**Comment by hejsiri** [comment]
A nie lepiej zrobić oddzielne kompilacje dla różnych języków? Tasmotę właśnie tak wypuszczają.

**Comment by bfaliszek** [comment]
Docelowo chcę tak zrobić i po zmianie w ustawieniach pobierałoby odpowiedni .bin z serwera i instalowało. Tylko najpierw chciałem wypuścić to co mam. Dodać automatyczne aktualizacje i później zacząć optymalizować i dzielić całość na mniejsze części. Żeby zrobić całość tak jak chcę, czyli kilkanaście mniejszych .bin w każdej kombinacji miernika pm z temp/humi i językami, muszę mieć działające automatyczne aktualizacje.

**Comment by ookris** [comment]
Opcja osobnej kompilacji dla każdego z języków to chyba dobry pomysł. 
@mientki strona z odczytami działa poprawnie bo niewiele jest tam do tłumaczenia.
